pkg/controllermanager/controller/project/project_activity_control.go

```diff
@@ -17,89 +17,199 @@ package project
 	}
 }
 
 type projectActivityReconciler struct {
 	gardenClient client.Client
+	clock        clock.Clock
 }
 
 func (r *projectActivityReconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
 	log := logf.FromContext(ctx)
 
-	shoot := &gardencorev1beta1.Shoot{}
-	if err := r.gardenClient.Get(ctx, client.ObjectKey{Namespace: request.Namespace, Name: request.Name}, shoot); err != nil {
+	project := &gardencorev1beta1.Project{}
+	if err := r.gardenClient.Get(ctx, request.NamespacedName, project); err != nil {
 		if apierrors.IsNotFound(err) {
 			log.Info("Object is gone, stop reconciling")
 			return reconcile.Result{}, nil
 		}
 		return reconcile.Result{}, fmt.Errorf("error retrieving object from store: %w", err)
 	}
 
-	return r.reconcile(ctx, log, shoot)
+	now := r.clock.Now()
+	log.Info("Updating Project's lastActivityTimestamp", "lastActivityTimestamp", now)
+	if err := updateStatus(ctx, r.gardenClient, project, func() {
+		project.Status.LastActivityTimestamp = &metav1.Time{Time: now}
+	}); err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed updating Project's lastActivityTimestamp: %w", err)
+	}
+
+	return reconcile.Result{}, nil
 }
 
-func (r *projectActivityReconciler) reconcile(ctx context.Context, log logr.Logger, obj client.Object) (reconcile.Result, error) {
-	project, err := gutil.ProjectForNamespaceFromReader(ctx, r.gardenClient, obj.GetNamespace())
+func (c *Controller) projectActivityObjectAddDelete(ctx context.Context, obj interface{}, withLabel bool, addFunc bool) {
+	objMeta, err := meta.Accessor(obj)
 	if err != nil {
-		if apierrors.IsNotFound(err) {
-			// project is gone, nothing to do here
-			return reconcile.Result{}, nil
-		}
-		return reconcile.Result{}, fmt.Errorf("failed to get project for namespace %q: %w", obj.GetNamespace(), err)
+		return
 	}
 
-	if project.Status.LastActivityTimestamp != nil && obj.GetCreationTimestamp().UTC().Before(project.Status.LastActivityTimestamp.UTC()) {
-		// not the newest object in this project, nothing to do
-		return reconcile.Result{}, nil
+	if withLabel {
+		// skip queueing if the object(secret or quota) doesn't have the "referred by a secretbinding" label
+		if _, hasLabel := objMeta.GetLabels()[v1beta1constants.LabelSecretBindingReference]; !hasLabel {
+			return
+		}
 	}
 
-	timestamp := obj.GetCreationTimestamp()
-	if project.Status.LastActivityTimestamp == nil {
-		shoots := &gardencorev1beta1.ShootList{}
-		if err := r.gardenClient.List(ctx, shoots, client.InNamespace(*project.Spec.Namespace)); err != nil {
-			return reconcile.Result{}, fmt.Errorf("failed to list shoots in project namespace %q: %w", *project.Spec.Namespace, err)
-		}
-		for _, shoot := range shoots.Items {
-			if shoot.GetCreationTimestamp().UTC().After(timestamp.UTC()) {
-				timestamp = shoot.GetCreationTimestamp()
-			}
+	if addFunc {
+		// If the creationTimestamp of the object is less than 1 hour (experimental) from current time,
```

Comment:
What does "experimental" mean here?

Reply:
Ok, maybe it was not clear, I meant this period of 1hr is experimental, and we can decrease or increase it after observing it for sometime.

Reply:
Yeah, probably it makes sense to remove this word from the comment to prevent such confusion. We can observe it and simply act if needed, i.e. no need to keep it here in the comment I think